rdmo/projects/utils.py

```diff
@@ -38,6 +39,86 @@ def check_conditions(conditions, values, set_prefix=None, set_index=None):
         return True
 
 
+def copy_project(project, site, owners):
+    from .models import Membership, Value  # to prevent circular inclusion
+
+    timestamp = now()
+
+    tasks = project.tasks.all()
+    views = project.views.all()
+
+    values = project.values.filter(snapshot=None)
+    snapshots = {
+        snapshot: project.values.filter(snapshot=snapshot)
+        for snapshot in project.snapshots.all()
+    }
+
+    # create a temporary buffer for all values with files
+    file_values = []
+
+    # unset the id, set current site and update timestamps
+    project.id = None
+    project.site = site
+    project.created = timestamp
+
+    # save the new project
+    project.save()
+
+    # save project tasks
+    for task in tasks:
+        project.tasks.add(task)
+
+    # save project views
+    for view in views:
+        project.views.add(view)
+
+    # save current project values
+    project_values = []
+    for value in values:
+        value.id = None
+        value.project = project
+        value.created = timestamp
+
+        if value.file:
+            file_values.append((value, value.file_name, value.file))
```

Comment:
for fixing failing test with mysql
```suggestion
        if value.file
            value.copy_file(value.file_name, value.file)
        else:
            project_values.append(value)
```

Reply:
This is actually much better, I refactored again and found a few bugs in the `rdmo/projects/tests/test_utils.py::test_copy_project` test.